src/main/java/org/spongepowered/common/mixin/api/mcp/world/raid/Raid_StatusMixin_API.java

```diff
@@ -0,0 +1,37 @@
+package org.spongepowered.common.mixin.api.mcp.world.raid;
+
+import org.spongepowered.api.CatalogKey;
+import org.spongepowered.api.data.type.RaidStatus;
+import org.spongepowered.api.plugin.PluginContainer;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.common.SpongeImplHooks;
+import org.spongepowered.common.bridge.CatalogKeyBridge;
+
+@Mixin(targets = "net/minecraft/world/raid/Raid$Status")
+public abstract class Raid_StatusMixin_API implements RaidStatus, CatalogKeyBridge {
+
+    @Inject(at = @At("HEAD"), method = "<init>()V")
+    private void impl$setCatalogKey(CallbackInfo ci) {
+        final PluginContainer container = SpongeImplHooks.getActiveModContainer();
+        this.impl$key = CatalogKey.of(container, this.shadow$func_221277_a());
+    }
+
+    @Shadow public abstract String shadow$func_221277_a();
+
+    private CatalogKey impl$key;
```

Comment:
If this field is added in the `api` package, it should be prefixed with `api$` not `impl$`, `impl$` belongs to `mixin.core`.